src/pkg/caendr/caendr/models/job_pipeline/indel_primer_pipeline.py

```diff
@@ -52,20 +65,141 @@ def parse(cls, data, valid_file_extensions=None):
     }
 
 
+
+  #
+  # Fetching & Parsing Input
+  #
+
+  def _parse_input(self, blob):
+
+    data = download_blob_as_json(blob)
+
+    # If data file is empty, raise an error
+    if data is None or not bool(data):
+      raise EmptyReportDataError(self.report.id)
+
+    return data
+
+
+
   #
-  # File Storage
+  # Fetching & Parsing Output
   #
 
-  def upload(self, data_files):
 
-    # Indel Finder only takes one upload file
-    if len(data_files) > 1:
-      raise ValueError('Only one data file should be uploaded.')
+  def fetch_output(self, raw: bool = False):
```

Comment:
Moved from view function, so that from the caller's perspective this is handled implicitly